electric_field/electric_field.js

```diff
@@ -50,6 +51,127 @@ class Particle {
   }
 
 }
+*/
+
+//Particle classs, draggable
+
+class Particle {
+  constructor(RAD) {
+    this.rad = RAD;
+
+    this.dragging = false; // Is the object being dragged?
+    this.rollover = false; // Is the mouse over the ellipse?
+
+    this.bs = 30;
+
+    if (random(-1, 1) < 0) {
+      this.label = "-";
+      this.mass = 9.109382e-31;
+      this.charge = -1.602176e-19;
+    } else {
+      this.label = "+";
+      this.mass = 1.672621e-27;
+      this.charge = 1.602176e-19;
+    }
+
+    this.pos = new Complex(random(RAD * 2, width - RAD * 2), random(RAD * 2, height - RAD * 2));
+
+  }
+
+  over() {
+    // Is mouse over object
+    if (mouseX > this.pos.re - this.bs && mouseX < this.pos.re + this.bs &&
+      mouseY > this.pos.im - this.bs && mouseY < this.pos.im + this.bs) {
+      this.rollover = true;
+    } else {
+      this.rollover = false;
+    }
+
+  }
+
+  update() {
```

Comment:
This function is not used. Please, call it inside `show()` or before it at the end of `draw`